packages/cli/src/databases/dsl/Table.ts

```diff
@@ -0,0 +1,82 @@
+import type { TableForeignKeyOptions, TableIndexOptions } from 'typeorm';
+import { Table, QueryRunner } from 'typeorm';
+import LazyPromise from 'p-lazy';
+import { Column } from './Column';
+
+abstract class TableOperation<R = void> extends LazyPromise<R> {
+	abstract execute(queryRunner: QueryRunner): Promise<R>;
+
+	constructor(
+		protected tableName: string,
+		protected prefix: string,
+		queryRunner: QueryRunner,
+	) {
+		super((resolve) => {
+			void this.execute(queryRunner).then(resolve);
+		});
+	}
+}
+
+export class CreateTable extends TableOperation {
+	private columns: Column[] = [];
+
+	private indicesOn = new Set<TableIndexOptions>();
+
+	private foreignKeys = new Set<TableForeignKeyOptions>();
+
+	withColumns(...columns: Column[]) {
+		this.columns.push(...columns);
+		return this;
+	}
+
+	get withTimestamps() {
+		this.columns.push(
+			new Column('createdAt').timestamp(3).notNull.default('NOW()'),
+			new Column('updatedAt').timestamp(3).notNull.default('NOW()'),
+		);
+		return this;
+	}
+
+	withIndexOn(columnName: string | string[], isUnique = false) {
+		const columnNames = Array.isArray(columnName) ? columnName : [columnName];
+		this.indicesOn.add({ columnNames, isUnique });
+		return this;
+	}
+
+	withForeignKey(
+		columnName: string,
+		ref: { tableName: string; columnName: string; onDelete?: 'CASCADE'; onUpdate?: 'CASCADE' },
```

Comment:
`'NO ACTION'`? `'SET NULL'`?

Reply:
this PR is a subset of a larger DSL I'm working on. Once we need those options, we can add those options. For now, this current code should be fine.